resiliency/service-to-service-resiliency.md

```diff
@@ -0,0 +1,159 @@
+```
+
+##### Checkout Service: 
+```bash
+cd ../service_invocation/java/http/checkout
+mvn clean install
+dapr run --app-id checkout --config ../config.yaml --components-path ../../../components/ --app-protocol http --dapr-http-port 3500 -- java -jar target/CheckoutService-0.0.1-SNAPSHOT.jar
+```
+
+### JavaScript example:
+##### Order Processor Service: 
+```bash
+cd ../service_invocation/javascript/http/order-processor
+npm install
+dapr run --app-port 5001 --app-id order-processor --app-protocol http --dapr-http-port 3501 -- npm start
+```
+
+##### Checkout Service: 
+```bash
+cd ../service_invocation/javascript/http/checkout
+npm install
+dapr run  --app-id checkout --config ../config.yaml --components-path ../../../components/ --app-protocol http --dapr-http-port 3500 -- npm start
+```
+
+### Python example:
+##### Order Processor Service: 
+```bash
+cd ../service_invocation/python/http/order-processor
+pip3 install -r requirements.txt 
+dapr run --app-port 8001 --app-id order-processor --app-protocol http --dapr-http-port 3501 -- python3 app.py
+```
+
+##### Checkout Service: 
+```bash
+cd ../service_invocation/python/http/checkout
+pip3 install -r requirements.txt 
+dapr run  --app-id checkout --config ../config.yaml --components-path ../../../components/ --app-protocol http --dapr-http-port 3500 -- python3 app.py
+```
+
+### Expected output: 
+Once both services are running, the `order-processor` service will recieve orders from the `checkout`service continuously using Dapr's service invoke API.
+
+##### `order-processor` output:
+```bash
+== APP == Order received: { orderId: 1 }
+== APP == Order received: { orderId: 2 }
+== APP == Order received: { orderId: 3 }
+```
+
+### Simulate ann application failure by stopping the `order-processor` service  
+
+Simulate a system failure by stopping the `order-processor` service running in your second terminal shell.
+
+##### Windows 
+```script
+CTRL + C
+```
+
+##### Mac
+```script
+CMD + C
+```
+
+### Observe retry and circuit breaker policies are applied:
+
+Policies defined in the resiliency.yaml spec:
+```yaml
+retryForever:
+  policy: constant
+  maxInterval: 5s
+  maxRetries: -1 
+
+circuitBreakers:
+  simpleCB:
+  maxRequests: 1
+  timeout: 5s 
+  trip: consecutiveFailures >= 5
+```
+
+##### Applied policies:
+```bash
+INFO[0005] Error processing operation endpoint[order-processor, order-processor:orders]. Retrying...  
+INFO[0025] Circuit breaker "order-processor:orders" changed state from closed to open  
+INFO[0030] Circuit breaker "order-processor:orders" changed state from open to half-open  
+INFO[0030] Circuit breaker "order-processor:orders" changed state from half-open to open  
+```
+
+### Simulate the application recovering by restarting the order-processor service:
+Simulate the `order-processor` service recovering by restarting the application using the `dapr run` command.
```

Comment:
Could we please spell out the entire `dapr run ..` command so everything about the quick start is copy/paste "drop in" easy?

Reply:
Done